File: src/dataset/Java/triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_Triangle_v2.java
public class Triangle {
    private double a, b, c;

    public Triangle(double a, double b, double c) throws TriangleException {
        if (a <= 0 || b <= 0 || c <= 0) {
            throw new TriangleException();
        }

        if (a + b <= c || a + c <= b || b + c <= a) {
            throw new TriangleException();
        }

        this.a = a;
        this.b = b;
        this.c = c;
    }

    public TriangleKind getKind() {
        if (a == b) {
            if (b == c) {
                return TriangleKind.EQUILATERAL;
            } else {
                return TriangleKind.ISOSCELES;
            }
        } else if (b == c || a == c) {
            return TriangleKind.ISOSCELES;
        } else {
            return TriangleKind.SCALENE;
        }
    }
}

File: src/dataset/python/triangle_exercism-ThomasZumsteg/gemini/GeminiFlash_triangle_v2.py
"""Validate and specify triangle types"""

class TriangleError(Exception):
    """Error if it cannot be a triangle"""
    def __init__(self, message):
        super().__init__(message)

class Triangle(object):
    """Validate and specify triangle types"""
    def __init__(self, a, b, c):
        """Create and validate triangle"""
        sides = sorted((a, b, c))
        self.a, self.b, self.c = sides[0], sides[1], sides[2]

        if self.a + self.b <= self.c:
            raise TriangleError("Not a Triangle")

    def kind(self):
        """Specify triangle type"""
        if self.a == self.b:
            if self.b == self.c:
                return "equilateral"
            else:
                return "isosceles"
        elif self.b == self.c:
            return "isosceles"
        else:
            return "scalene"